Sources/ApolloCodegenLib/ApolloCodegenConfiguration.swift

```diff
@@ -0,0 +1,140 @@
+import Foundation
+
+/// A configuration object that defines behavior for code generation.
+public struct ApolloCodegenConfiguration {
+  /// Specify the input files required for code generation.
+  public struct FileInput {
+    /// Path to the GraphQL schema file. Can be in JSON or SDL format.
+    public let schema: URL
+    /// Glob of files to search for GraphQL operations. This should be used to find queries and any client schema extensions. Defaults to `./**/*.graphql`, which will search for `.graphql` files throughout all subfolders of the folder where the script is run.
```

Comment:
Nit: If documentation lines are longer than 100 character line limit, I've been pulling them to a second line.
```swift
///  Glob of files to search for GraphQL operations. This should be used to find queries and any client schema extensions. Defaults to `./**/*.graphql`,
/// which will search for `.graphql` files throughout all subfolders of the folder where the script is run.
```

What's your opinion on that as a style guideline?

Reply:
Personally I'm fine with wrapping docs because they're auto-wrapped by the IDE, but it is a pain to read them in GH so fine if y'all want to do that.

Reply:
(although split mode in GH does auto wrap)

Reply:
I have my page guide set to 100 so yes I should have caught this; I'll break this and others.

On the whole I think the [Google Swift style guide](https://google.github.io/swift/) is pretty good and take direction from that generally. I believe it would form a good foundation for a [swift-format](https://github.com/apple/swift-format) and [SwiftLint](https://github.com/realm/SwiftLint) rule set.

Reply:
Yeah we've been meaning to add swiftlint for...some time